src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -682,7 +692,7 @@ void ProcessOverride (OverrideInformation overrideInformation)
 			// Only track instantiations if override removal is enabled and the type is instantiated.
 			// If it's disabled, all overrides are kept, so there's no instantiation site to blame.
 			if (_context.IsOptimizationEnabled (CodeOptimizations.OverrideRemoval, method) && isInstantiated) {
-				MarkMethod (method, new DependencyInfo (DependencyKind.OverrideOnInstantiatedType, method.DeclaringType), null);
+				MarkMethod (method, new DependencyInfo (DependencyKind.OverrideOnInstantiatedType, method.DeclaringType), new MessageOrigin (method.DeclaringType));
```

Comment:
Could we use the stack here as well so that we get rid of the last parameter (origin) in MarkMethod?

Reply:
Not really - there's nothing on the stack when this is called - this is because virtual override processing is delayed and done outside of normal marking (from `ProcessVirtualMethods`). We could setup something onto the stack, but the most natural thing would the method itself. It's unclear if that's better scope that the type. Since virtual methods are largely marked because of their type (it's instantiated and so on).

Reply:
And I immediately prove myself wrong.
Turns out we have to propagate the scope from the `ProcessMethod` to these delayed processing queues, otherwise the upcoming compiler generated code handling would not work (since it needs a parent scope at all times).

So you were right.